### PR TITLE
Add kwargs for gateway IP, DNS, CPU/memory

### PR DESCRIPTION
**Pull request by omyno**
### What does this PR do?
Adds kwargs for gateway IP, DNS, CPU/memory

### What issues does this PR fix or reference?
-

### Previous Behavior
Settings for gateway IP etc. will be ignored

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rallytime** [comment]
Welcome @omyno!

Can you fix these lint errors? https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/20462/violations/file/salt/cloud/clouds/dimensiondata.py/

File: salt/cloud/clouds/dimensiondata.py [hunk 1]
```diff
@@ -248,7 +248,13 @@ def create(vm_):
         'ex_description': vm_['description'],
         'ex_network_domain': network_domain,
         'ex_vlan': vlan,
-        'ex_is_started': vm_['is_started']
+        'ex_is_started': vm_['is_started'],
+        'ex_ipv4_gateway': vm_['ipv4_gateway'],                     
+        'ex_primary_dns': vm_['primary_dns'],
+        'ex_secondary_dns': vm_['secondary_dns'],
```
Comment (cachedout) [inline salt/cloud/clouds/dimensiondata.py]: Are we certain that these values will always be set? I'm just wondering if we should use a `.get()` here instead.

**Comment by cachedout** [comment]
@omyno Did you see the above feedback on your PR?

File: salt/cloud/clouds/dimensiondata.py (hunk 1 above)
Comment (cachedout) [reply salt/cloud/clouds/dimensiondata.py]: So if they aren't mandatory do you agree we should change this so we don't stacktrace? :]

Comment (omyno) [reply salt/cloud/clouds/dimensiondata.py]: Yes, absolutely!

Comment (cachedout) [reply salt/cloud/clouds/dimensiondata.py]: Cool. If you'd like to go ahead and make those changes then we should be able to get this in. Thanks!

**Comment by rallytime** [comment]
Since we haven't heard back here for a while, I am going to close this. We would be happy to open this again when you're ready. Or please feel free to open a new PR. Whichever is easier for you. :)

Thanks!

**Pull request closed without merging**